File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.kt
class Solution {
    fun solution(board: Array<IntArray>, moves: IntArray): Int {
        var answer = 0
        val stack = mutableListOf<Int>()

        for(i in moves.indices) {
            for(j in board.indices) {
                if(board[j][moves[i]-1] != 0) {
                    stack.add(board[j][moves[i]-1])
                    board[j][moves[i]-1] = 0
                    if(stack.size > 1 && stack[stack.size-1] == stack[stack.size-2]) {
                        stack.removeAt(stack.size-1)
                        stack.removeAt(stack.size-1)
                        answer += 2
                    }
                    break
                }
            }
        }
        return answer
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.swift
func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    var answer = 0
    var stack = [Int]()
    var verticalBoard = Array(repeating: [Int](), count: board.count)
    
    for horizontal in board {
        for (index, doll) in horizontal.enumerated() {
            if doll != 0 {
                verticalBoard[index].append(doll)
            }
        }
    }
    
    for move in moves {
        if !verticalBoard[move-1].isEmpty {
            let popDoll = verticalBoard[move-1].remove(at: 0)
            stack.append(popDoll)
            
            let count = stack.count
            if count >= 2, stack[count-1] == stack[count-2] {
                stack.removeSubrange(count-2...count-1)
                answer += 2
            }
        }
    }
    
    return answer
}